demisto_sdk/commands/validate/tests/DO_validators_test.py

```diff
@@ -80,20 +69,506 @@ def test_DockerImageExistValidator_is_valid(
-        - Case 3: Should'nt fail at all.
+        - Case 3: Shouldn't fail at all.
         - Case 4: Should fail all content items.
     """
+    from demisto_sdk.commands.validate.validators.DO_validators.DO104_docker_image_does_not_exist_in_yml import (
+        DockerImageExistValidator,
+    )
+
     mocker = mocker.patch.object(
-        DockerImageValidator,
-        "get_docker_image_latest_tag_request",
+        DockerHubClient,
+        "get_latest_docker_image_tag",
         return_value="3.1.1.1",
     )
     results = DockerImageExistValidator().is_valid(content_items)
-    len(results) == expected_number_of_failures
-    assert all(
-        [
-            result.message == expected_msg
-            for result, expected_msg in zip(results, expected_msgs)
-        ]
-    )
-    mocker.call_count == expected_call_count
+    assert len(results) == expected_number_of_failures
+    assert mocker.call_count == expected_call_count
+
+
+def test_LatestDockerImageTagValidator_is_valid():
+    """
+    Given:
+     - 1 integration and 1 script which uses the latest tag
+     - 1 integration and 1 script which do not use the 'latest' tag
+     - 1 integration and 1 script that are javascript
+
+    When:
+     - Running the LatestDockerImageTagValidator validator
+
+    Then:
+     - make sure the integrations/scripts with the "latest" tags are not valid
+     - make sure that javascripts integrations/scripts are ignored
+    """
+    from demisto_sdk.commands.validate.validators.DO_validators.DO100_docker_image_tag_is_not_latest import (
+        LatestDockerImageTagValidator,
+    )
+
+    content_items = [
+        create_integration_object(
+            paths=["name", "script.dockerimage"],
+            values=[
+                "LatestDockerImageIntegrationScript",
+                "demisto/python3:latest",
+            ],  # integration with 'latest' as the docker tag
+        ),
+        create_script_object(
+            paths=["name", "dockerimage"],
+            values=[
+                "LatestDockerImageIntegrationScript",
+                "demisto/python3:latest",
+            ],  # script with 'latest' as the docker tag
+        ),
+        create_integration_object(),  # integration without 'latest' docker tag
+        create_script_object(),  # script without 'latest' docker tag
+        create_integration_object(
+            paths=["script.type"], values=["javascript"]
+        ),  # javascript integration
+        create_script_object(
+            paths=["type"], values=["javascript"]
+        ),  # javascript script
+    ]
+    results = LatestDockerImageTagValidator().is_valid(content_items)
+    assert len(results) == 2
+    for result in results:
+        content_item: IntegrationScript = result.content_object
+        assert content_item.type == "python"
+        assert content_item.docker_image == "demisto/python3:latest"
+        assert content_item.name == "LatestDockerImageIntegrationScript"
```

Comment:
Let's validate the message as well.

Reply:
i don't think its a good practice to validate messages or strings in unit-tests, it makes them very "breakable" for no real good reason.